server/issue.go

```diff
@@ -209,6 +211,72 @@ func httpAPICreateIssue(ji Instance, w http.ResponseWriter, r *http.Request) (in
 	return http.StatusOK, nil
 }
 
+func httpWorkflowCreateIssue(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	if r.Method != http.MethodPost {
+		return http.StatusMethodNotAllowed,
+			errors.New("method " + r.Method + " is not allowed, must be POST")
+	}
+
+	var activationParams workflowclient.ActionActivationParams
+	if err := json.NewDecoder(r.Body).Decode(&activationParams); err != nil {
+		return http.StatusBadRequest,
+			errors.WithMessage(err, "failed to decode incoming request")
+	}
+
+	create := &struct {
+		UserId string           `json:"user_id"`
+		Fields jira.IssueFields `json:"fields"`
+	}{}
+	if err := json.NewDecoder(bytes.NewReader(activationParams.Action)).Decode(&create); err != nil {
+		return http.StatusBadRequest,
+			errors.WithMessage(err, "failed to decode incoming request specific parameters")
+	}
+
+	var client interface {
+		CreateIssue(issue *jira.Issue) (*jira.Issue, error)
+	}
+	if create.UserId != "" {
+		jiraUser, err := ji.GetPlugin().userStore.LoadJIRAUser(ji, create.UserId)
+		if err != nil {
+			return http.StatusInternalServerError, err
+		}
+
+		client, err = ji.GetClient(jiraUser)
+		if err != nil {
+			return http.StatusInternalServerError, err
+		}
+	} else {
+		jci, ok := ji.(*jiraCloudInstance)
+		if !ok {
+			return http.StatusBadRequest, errors.New("UserId is required for jira server instances.")
+		}
+
+		jiraClient, err := jci.getJIRAClientForServer()
```

Comment:
Do you mind renaming this function to something like `getJIRAClientForBot`? It's confusing because it says the word `Server`, though it's for Jira Cloud. I think this is as of a good a time as any to rename it.

Reply:
Done!